katsdpdata/met_extractors.py

```diff
@@ -395,454 +376,133 @@ def extract_metadata(self):
-        bandwidth=float(output.split(' ')[5])
-        centre_freq = output.split(' ')[6]
-        min_freq = (float(centre_freq) - bandwidth/2) * 1000000
-        max_freq = (float(centre_freq) + bandwidth/2) * 1000000
-        bandwidth=bandwidth * 1000000
-        
-        obs_info_file = open ("%s/obs_info.dat"%self.product_name)
-        obs_info = dict([a.split(';') for a in obs_info_file.read().split('\n')[:-1]])
-        self.metadata["Observer"]=obs_info["observer"]
-        self.metadata["Bandwidth"]=str(bandwidth)
-        self.metadata["MinFreq"]=str(min_freq)
-        self.metadata["MaxFreq"]=str(max_freq)
-        self.metadata["ProgramBlockId"]=obs_info["program_block_id"]
-        self.metadata["ScheduleBlockIdCode"]=obs_info["sb_id_code"]
-        self.metadata["Duration"]=obs_info["target_duration"]
-        self.metadata["ProposalId"]=obs_info["proposal_id"]
-        self.metadata["Description"]=obs_info["description"]
-        self.metadata["ExperimentID"]=obs_info["experiment_id"]
-        self.metadata["CAS.ProductTypeName"]='PTUSETimingArchiveProduct'
-        self.metadata["ScheduleBlockIdCode"]=obs_info["sb_id_code"]
-        self.metadata['Description'] = obs_info["description"]
-        self.metadata['FileSize'] = str(sum(os.path.getsize(f) for f in os.listdir(self.product_name) if os.path.isfile(f)))
-        self.metadata['KatfileVersion'] = "ar"
-        import ast
-        self.metadata['Targets'] = ast.literal_eval(obs_info["targets"])
-        self.metadata['StartTime'] = startTime
-
-        import katpoint
-        t = katpoint.Target("%s, radec, %s, %s"%(self.metadata["Targets"][0],radec[0],radec[1]))
-        self.metadata['KatpointTargets'] = [t.description]
-        self._metadata_extracted = True
-
-
-class CalibrationProductMetExtractor(MetExtractor):
-    """A class for handling calibration report metadata extraction.
+def file_mime_detection(katfile):
+    """Fuction to instantiate the correct metadata extraction class. The
+    following file extentions are are currently detected: '.h5' and '.rdb'.
 
-    Parameters
+    Parameters:
     ----------
-    prod_name : string : the name of a heirachical product to ingest.
+    katfile: string : name of file for mime detecetion.
     """
-    def __init__(self, prod_name):
-        super(CalibrationProductMetExtractor, self).__init__("%s.%s"%(prod_name,"met"))
-        self.product_type = 'MeerKATAR1ReductionProduct'
-        self.product_name = prod_name
-
-    def extract_metadata(self):
-        if not self._metadata_extracted:
-            self._extract_metadata_product_type()
-            self.metadata['ReductionName'] = "Calibration Report"
-            self.metadata['CaptureBlockId'] = self.product_name.split('/')[-1][:10]
-            self._metadata_extracted = True
+    file_ext = os.path.splitext(katfile)[1]
+    if file_ext == '.h5':
+        katdata = katdal.open(katfile)
+        # atleast one antenna starts with 'ant'
```

Comment:
Good point. I've updated the comments.

Reply:
So ants are sorted list  (alphabetically). By chance meerkat antennas start with "m", so "ant" will always be first in the list. Since this legacy code, I'm leaving it as is, but I've made the comment more detailed. This lead me down some other paths... seems like I'm still not extracting the Instrument from .rdb files correctly! I now make sure that SITENAME gets extracted as Instrument for Chris Schollar to use.